ecdsa/src/lib.rs

```diff
@@ -672,11 +674,11 @@ impl<C> DynAssociatedAlgorithmIdentifier for SignatureWithOid<C>
 where
     C: PrimeCurve,
 {
-    fn algorithm_identifier(&self) -> AlgorithmIdentifierOwned {
-        AlgorithmIdentifierOwned {
+    fn algorithm_identifier(&self) -> SpkiResult<AlgorithmIdentifierOwned> {
```

Comment:
```suggestion
    fn algorithm_identifier(&self) -> spki::Result<AlgorithmIdentifierOwned> {
```

Reply:
@tarcieri there is no direct dependency on `spki`. Would you prefer:
- add spki as a dep?
- use whatever..spki to make `spki` available as a name
- use full path including elliptic-curves::pkcs8:...

Reply:
See: https://github.com/RustCrypto/signatures/pull/695/files#r1157862631